presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -619,6 +625,12 @@ PrestoServer::getHttpServerFilters() {
             httpServer_.get()));
   }
 
+  if (SystemConfig::instance()->internalCommunicationJwtEnabled() &&
```

Comment:
The code is a result of a code review comment by @majetideepak (https://github.com/prestodb/presto/pull/20290#discussion_r1302125425). Initially, the define was outside.
The config is available but is not active if not built (regardless of whether or not the check occurs here or is ifdef'ed out). It is one of these things. 
This might be the first time a config exists for an optional (default off - see above) feature. We can add a warning to the presto startup log if configured but not built in.

Reply:
I meant to move the check inside the function `internalCommunicationJwtEnabled()`. But that will not handle the body.
Let's keep 
```
#ifdef PRESTO_ENABLE_JWT
  if (SystemConfig::instance()->internalCommunicationJwtEnabled()) {
    filters.push_back(std::make_unique<http::filters::JWTTokenFilterFactory>());
     PRESTO_STARTUP_LOG(INFO)
        << "Internal communication using JWT authentication is configured.";
 }
#else
```